packages/client/vite.config.ts

```diff
@@ -70,51 +70,48 @@ function targetBuildPlugin(dev: boolean, target: string): Plugin {
         const mockPath = `${file.dir}/${file.name}.${target}.${extension}`
         return this.resolve(mockPath, importer)
       } else {
-        const rootPrefix = "src/client/"
-        const thisImporter = (importer || "").replace(/\\/g, "/")
         if (
           !importer ||
-          !thisImporter.includes(rootPrefix) ||
+          importer.includes("node_modules") ||
           source === "./main"
         ) {
           return null
         }
 
-        const importedFile = path.parse(source)
+        if (!source.startsWith(".") && !source.startsWith("/")) {
+          return null
+        }
+
+        const sourcePath = path.parse(source)
```

Comment:
by the time we get here, we are down to 865 genuine cases of imports that could be switched to target-specific replacements .. we may be able to filter further, but this seems reasonable now